From 227619d1abe26436c98f022ee81bd5d4efb74dd2 Mon Sep 17 00:00:00 2001
From: Sanyog Kale <skale@habana.ai>
Date: Thu, 01 Sep 2022 09:42:51 +0300
Subject: [PATCH] [SW-102303] gaudi3: fix register offset for cpld wd timer

This patch fixes offset of GPIO register in
arcboot_rom_start_cpld_wdtimer to indicate start of watchdog timer
to CPLD.

Change-Id: I4748e66ed3dbc9ca2b791d5dcbdcf465b52fadc0
---

diff --git a/arch/gaudi3/arc/common.c b/arch/gaudi3/arc/common.c
index 6de44dd..9327cac 100644
--- a/arch/gaudi3/arc/common.c
+++ b/arch/gaudi3/arc/common.c
@@ -587,9 +587,9 @@
 	regmap_writel(val, mmD0_PSOC_GPIO4_OUTENSET);
 
 	/* gpio is active low */
-	val = regmap_readl(mmD0_PSOC_GPIO4_OUTENSET);
+	val = regmap_readl(mmD0_PSOC_GPIO4_DATAOUT);
 	val &= ~mask;
-	regmap_writel(val, mmD0_PSOC_GPIO4_OUTENSET);
+	regmap_writel(val, mmD0_PSOC_GPIO4_DATAOUT);
 }
 
 static void arcboot_get_boot_desc(int img, struct boot_desc *desc, enum fw_post_reset_cmd upd_status)
